buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlRestCompatTestPlugin.java

```diff
@@ -73,84 +74,54 @@ public void apply(Project project) {
         ElasticsearchCluster testCluster = createTestCluster(project, yamlCompatTestSourceSet);
         testCluster.setTestDistribution(TestDistribution.DEFAULT);
 
-        // TODO: once https://github.com/elastic/elasticsearch/pull/62473 lands refactor this to reference the checkoutDir as an artifact
-        int priorMajorVersion = VersionProperties.getElasticsearchVersion().getMajor() - 1;
-        final Path checkoutDir = project.findProject(":distribution:bwc:minor")
-            .getBuildDir()
-            .toPath()
-            .resolve("bwc")
-            .resolve("checkout-" + priorMajorVersion + ".x");
-
         // copy compatible rest specs
-        Configuration compatSpec = project.getConfigurations().create("compatSpec");
-        Configuration xpackCompatSpec = project.getConfigurations().create("xpackCompatSpec");
-        Configuration additionalCompatSpec = project.getConfigurations().create("additionalCompatSpec");
+        Configuration bwcMinorConfig = project.getConfigurations().create("bwcMinor");
+        Dependency bwcMinor = project.getDependencies().project(Map.of("path", ":distribution:bwc:minor", "configuration", "checkout"));
```

Comment:
I forget what the plan is for expanding this beyond just hard-coding this to only test against minor. I think we need some infra for being able to download rest test artifacts but we could also do test against "staged" right now.  I think a better intermediare step would be to use `BwcVersions.forPreviousUnreleased` here to iterate over any existing unreleased BWC branches.

Reply:
The plan is to (manually) deploy zip artifacts that include all of the tests and apis that are found in source. Then we can iterate over all of the latest of bugfix version per minor.  We will also want to iterate over the latest bugfix and staged too, which will need to be sourced from git just like here. 

I am hesitant to introduce that right now because I think all previous released and bugfix/staged would live under a different lifecycle task (which can be introduced later). The tests against BWC minor are wired in via `check` so that normal YAML tests will run, followed by the compatible YAML tests. I believe that we need to keep at least one compatible test as part of the primary check per project and for now it is bwc:minor , but in future (for performance reasons) we may want to consider to be previous released minor.  

I think it will all get there before release, but holding off a little bit on the introduction of the new lifecycle allow me to focus on a few other things first.

Reply:
The way we handle BWC checks wrt lifecycle tasks now is that we have a `bwcTestSnapshots` which is wired to `check` and this task tests only snapshot bwc versions, so in this case for `master` that would be `7.x`, `7.10` and `7.9` (until it goes GA). All release versions are tested explicity and not when running `check`. I think we want to continue with this pattern of `check` running all active dev code.

Reply:
I think the primary difference here is that this will be applied for each and every project, as opposed to a special purpose project. For example, I make a change to the reindex module, do we want to have to checkout 3 branches of source code and run 3 additional test clusters ? (same applies for every module/plugin)

Today, most dev workflows don't result in executing the bwcTestSnapshots, resulting in CI to run the that special purpose project(s). I think we should introduce a new task(s) to run the full compat set of tests that will be run via CI, but we need to be careful with how much additional testing per (normal) project. 

I think my comfort level is 1 additional set of REST tests per project that are executed via :myproject:check, and rely on CI job(s) for the rest of the compatibility. I believe that testing against bwc:minor or the latest release will be sufficient for the majority of the cases and can rely on special CI jobs for the long tail.

Reply:
Ah, I see what you are saying. That being said, I think even effectively doubling the rest tests across the board is a bit much since the vast majority of changes are unlikely to impact REST compatibility. I think in terms of task lifecycle we probably want to completely decouple this stuff from `check` or introduce some other more lightweight variant for folks to run locally. We'll also most likely end up splitting all this into it's own CI jobs.

Do we have runtimes on this stuff?

Reply:
The runtimes will be approximately equal to the normal yamlRestTests. 

for example: https://gradle-enterprise.elastic.co/s/p65q5te3ybeni/tests?search=yamlRestTest (part1)
and https://gradle-enterprise.elastic.co/s/72gyujegdjpp2/tests?search=yamlRestTest (part2)

>  I think in terms of task lifecycle we probably want to completely decouple this stuff from check or introduce some other more lightweight variant for folks to run locally. 

I am good with that especially after double checking the runtimes.  I don't know of a way to make a light weight check that would be effective ... if we decouple with `check` we should probably introduce a `compatTestSnapshots` that runs the equivalent of `bwcTestSnapshots` (sourcing from the possible 3 git sources) in a separate CI job run per PR. Additionally we would want a compatTestAll that would not run per PR but likely a dedicated job. 

All good conversations, but abit outside the scope of this PR and can decouple the check and introduce separate lifecycle tasks in future PR.

Reply:
I wonder if we should consider these tests "bwc" tests for the purposes of the `bwc_tests_enabled` flag. We already have [some logic to separate out bwc testing into its own job](https://github.com/elastic/elasticsearch/blob/27089e74602dc442656cf97ca4b29af79f724933/build.gradle#L188-L191), as well as allow devs to disable bwc testing when in the process of backporting BWC changes. We don't necessarily have to do that now though, we can leave this as a TODO.